src/codegen/revert.rs

```diff
@@ -77,8 +93,28 @@ impl SolidityError {
     ) -> Option<Expression> {
         match self {
             Self::Empty => None,
-            Self::String(data) => {
-                let args = vec![self.selector_expression(), data.clone()];
+            Self::Error(expr) => {
+                let args = vec![self.selector_expression(ns), expr.clone()];
+                create_encoder(ns, false)
+                    .const_encode(&args)
+                    .map(|bytes| {
+                        let size = Expression::NumberLiteral {
+                            loc: Codegen,
+                            ty: Type::Uint(32),
```

Comment:
`Type::Bytes(4)` would be much nicer imho

Reply:
I'm not sure. Why do you think that the buffer size not be an uint?

Reply:
Sorry, don't know what I was thinking. I was confused with the selector here, ignore me